### ISSUE TITLE
[Proposal] Add MLP transcoders

### ISSUE THREAD

**Issue by dtch1997** [jbloomAus/SAELens#182]
### Proposal 

Support training, loading, and inference of MLP transcoders.  

### Motivation

MLP transcoders were trained by Jacob Dunefsky and Philippe Chlenski and have been shown to be useful. 

### Pitch

- Implement a `HookedTranscoder` class analogous to `HookedSAE` and using similar functionality. 
- Implement a transcoder training runner. 
- Support loading pre-trained transcoder checkpoints.

### Checklist

- [X] I have checked that there is no similar [issue](https://github.com/jbloomAus/SAELens/issues) in the repo (**required**)

**Comment by dtch1997** [comment]
I'm likely going to be rather hacky with the first pass of the implementation, and possibly duplicate a bunch of code in order to have maximum freedom to make changes. Refactoring can be done later.

**Comment by anthonyduong9** [comment]
@dtch1997 are you planning on requesting a review on https://github.com/jbloomAus/SAELens/pull/183? If not, I'd like to work on this issue/build on your PR!

**Comment by OmarMohammed88** [comment]
any updates regarding adding it ?

**Comment by anthonyduong9** [comment]
@OmarMohammed88 yeah, I chatted in person with @dtch1997 about this, and I plan to complete it, but after https://github.com/jbloomAus/SAELens/pull/425 is merged. I'll assign myself to the issue.

**Comment by shivam-raval96** [comment]
@anthonyduong9 Hi, my friend and I are also considering working in this direction. Would you be available to meet and chat in the coming days? Our emails are sraval@g.harvard.edu and anton_poletaev@outlook.com. Looking forward to chatting:)

**Comment by anthonyduong9** [comment]
@shivam-raval96 hi, interesting! I'd be available. Could you book a meeting at https://calendly.com/anthonyduong9?

**Comment by wujuu** [comment]
Hey @anthonyduong9, @shivam-raval96, @dtch1997  have you been able to get any work done regarding Transcoder support here?
